site/gatsby-site/src/components/SemanticallyRelatedIncidents.js

```diff
@@ -0,0 +1,164 @@
+import React, { useEffect, useState, useRef } from 'react';
+import { gql, useApolloClient } from '@apollo/client';
+import debounce from 'lodash/debounce';
+import RelatedIncidentsArea from './RelatedIncidentsArea';
+import { stripMarkdown } from '../utils/typography';
+import { useFormikContext } from 'formik';
+
+const relatedIncidentIdsQuery = gql`
+  query ProbablyRelatedIncidentIds($query: IncidentQueryInput) {
+    incidents(query: $query) {
+      incident_id
+      reports {
+        report_number
+        title
+        url
+      }
+    }
+  }
+`;
+
+const semanticallyRelated = async (text, max_tries) => {
+  const url = `/api/semanticallyRelated?text=${text}`;
```

Comment:
I think the retry logic should likely be in the API and hidden from the UI.

Reply:
Got it. I made the change in #736 and merged it here.